### PR TITLE
Add Live Activity widget to sample app

### PR DESCRIPTION
**Pull request by brismithers**
# Description
## One Line Summary
Add a Live Activity example to the sample app.

### Motivation
In order to showcase how a Xamarin app could produce a live activity, and have it be updated via the OneSignal platform.  Xamarin does not have support WidgetKit/ActivityKit, so native bindings and native extensions were created.  The general approach was adopted from the WidgetKit example provided by https://github.com/chamons/xamarin-ios-swift-extension.

Note that because this approach requires a native build of the widget in order for the Xamarin solution to build, the Live Activity functionality defaults to disabled.  It can be enabled by specifying the `LIVE_ACTIVITIES` compiler constant.  See `Samples/LIVE_ACTIVITIES.md` for more explicit instructions.

### Scope
This change should *only* affect the sample application.

# Testing

## Manual testing
The sample app was tested on both iPhone and iPhoneSimulator, with live activity functionality enabled and disabled.  Entering/exit live activity was driven in all variations to ensure correct behavior.

# Affected code checklist
   - [ ] Notifications
      - [ ] Display
      - [ ] Open
      - [ ] Push Processing
      - [ ] Confirm Deliveries
   - [ ] Outcomes
   - [ ] Sessions
   - [ ] In-App Messaging
   - [ ] REST API requests
   - [ ] Public API changes

# Checklist
## Overview
   - [X] I have filled out all **REQUIRED** sections above
   - [X] PR does one thing
     - If it is hard to explain how any codes changes are related to each other then it most likely needs to be more than one PR
   - [X] Any Public API changes are explained in the PR details and conform to existing APIs

## Testing
   - [X] I have included test coverage for these changes, or explained why they are not needed
   - [X] All automated tests pass, or I explained why that is not possible
   - [X] I have personally tested this on my device, or explained why that is not possible

## Final pass
   - [X] Code is as readable as possible.
      - Simplify with less code, followed by splitting up code into well named functions and variables, followed by adding comments to the code.
   - [X] I have reviewed this PR myself, ensuring it meets each checklist item
      - WIP (Work In Progress) is ok, but explain what is still in progress and what you would like feedback on. Start the PR title with "WIP" to indicate this.

- - -
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-xamarin-sdk/345)

### CODE REVIEW

**Review by jkasten2 — CHANGES_REQUESTED** [review]
> Reviewed 37 of 37 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/onesignal/onesignal-xamarin-sdk/345)* status: all files reviewed, 1 unresolved discussion (waiting on @brismithers and @fhboswell)

___
*[`.gitignore` line 7 at r1](https://reviewable.io/reviews/onesignal/onesignal-xamarin-sdk/345#-NIEETP42s-tTyhI93Iz:-NIEETP42s-tTyhI93J-:b1acqsd) ([raw file](https://github.com/onesignal/onesignal-xamarin-sdk/blob/b0180f5cc4f063ea869d48e6b95da07897dfe3b7/.gitignore#L7)):*
> ```ignore
> [Oo]bj/
> [Bb]in/
> DerivedData/
> ```

Normally you don't want to check this in, since it contains built binaries the build process will generate automatically.
I could see this being needed however based on the fact of how the project is being linked, however I see no new files in this PR added to to `DerivedData/` so this isn't needed.
* If you missed some files and this really is needed, this still is concerning, as I think DerivedData is Xcode version specific too, but could be wrong.

File: Samples/native/example/OneSignalLiveActivity/OneSignalLiveActivity.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        if #available(iOS 16.1, *) {
+            let attributes = OneSignalWidgetAttributes(title: "OneSignal Dev App Live Activity")
+            let contentState = OneSignalWidgetAttributes.ContentState(message: "Update this message through push or with Activity Kit")
+
+            do {
+                let activity = try Activity<OneSignalWidgetAttributes>.request(
+                    attributes: attributes,
+                    contentState: contentState,
+                    pushType: .token)
+                print(activity.id)
```
Comment (fhboswell) [inline Samples/native/example/OneSignalLiveActivity/OneSignalLiveActivity.swift:38]: nit: remove print statement

File: Samples/native/example/OneSignalWidget/OneSignalWidget.swift [hunk 2]
```diff
@@ -0,0 +1,68 @@
+//
+//  OneSignalWidget.swift
+//  OneSignalWidget
+//
+//  Created by Brian Smith on 11/30/22.
+//
+
+import WidgetKit
```
Comment (fhboswell) [inline Samples/native/example/OneSignalWidget/OneSignalWidget.swift:8]: This file is unused. This is for a (non-live activity) widget.

File: Samples/native/example/example/ContentView.swift [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+//  ContentView.swift
+//  example
+//
+//  Created by Brian Smith on 11/30/22.
+//
+
+import SwiftUI
+
+struct ContentView: View {
+    var body: some View {
```
Comment (fhboswell) [inline Samples/native/example/example/ContentView.swift:11]: Is this file for the host app for the live activity?

**Review by fhboswell — COMMENTED** [review]
A couple questions and comments.

**Review by fhboswell — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/onesignal/onesignal-xamarin-sdk/345)* status: all files reviewed, 4 unresolved discussions (waiting on @brismithers and @jkasten2)

___
*[`.gitignore` line 7 at r1](https://reviewable.io/reviews/onesignal/onesignal-xamarin-sdk/345#-NIEETP42s-tTyhI93Iz:-NIEo3fbCiP-v_FSA7Hu:b-spdax9) ([raw file](https://github.com/onesignal/onesignal-xamarin-sdk/blob/b0180f5cc4f063ea869d48e6b95da07897dfe3b7/.gitignore#L7)):*
<details><summary><i>Previously, jkasten2 (Josh Kasten) wrote…</i></summary><blockquote>

Normally you don't want to check this in, since it contains built binaries the build process will generate automatically.
I could see this being needed however based on the fact of how the project is being linked, however I see no new files in this PR added to to `DerivedData/` so this isn't needed.
* If you missed some files and this really is needed, this still is concerning, as I think DerivedData is Xcode version specific too, but could be wrong.
</blockquote></details>

After seeing a demo it is my understanding that this directory is only used to hold the build output of the Xcode project which hosts the live activity, and therefore it makes sense to gitignore it.

**Review by jkasten2 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/onesignal/onesignal-xamarin-sdk/345)* status: all files reviewed, 3 unresolved discussions (waiting on @brismithers)

___
*[`.gitignore` line 7 at r1](https://reviewable.io/reviews/onesignal/onesignal-xamarin-sdk/345#-NIEETP42s-tTyhI93Iz:-NIEsPYWEDxYCODEsLQu:b-zdn4rg) ([raw file](https://github.com/onesignal/onesignal-xamarin-sdk/blob/b0180f5cc4f063ea869d48e6b95da07897dfe3b7/.gitignore#L7)):*
<details><summary><i>Previously, fhboswell (Franklin Henry Boswell) wrote…</i></summary><blockquote>

After seeing a demo it is my understanding that this directory is only used to hold the build output of the Xcode project which hosts the live activity, and therefore it makes sense to gitignore it.
</blockquote></details>

I don't know why but I thought this was a delete rather than an add, disregard my first comment. No issue here.

File: Samples/native/example/example/ContentView.swift (hunk 3 above)
Comment (brismithers) [reply Samples/native/example/example/ContentView.swift:11]: This file is for the "dummy host app" it is required for the Xamarin solution to properly pull in the widget extension.  It is not actually used.

**Comment by brismithers** [comment]
@jkasten2 regarding the `DerivedData` in the `.gitignore` file.  In order for the widget to work you must open the xcode project and build to `DerivedData`, which is then referenced by the Xamarin project.  And in order for the LiveActivity portion of the sample app to work you have to go through some steps (1 of them being to do an xcode build into `DerivedData`).  I include it in the `.gitignore` file because I don't want it accidentally checked in.  Let me know if this makes sense!

File: Samples/native/example/OneSignalWidget/OneSignalWidget.swift (hunk 2 above)
Comment (brismithers) [reply Samples/native/example/OneSignalWidget/OneSignalWidget.swift:8]: Yup!  I kept that in there to help through the setup process as it was easier to get the widget showing than the live activity.  It is included in the widget bundle for this purpose.

**Review by fhboswell — APPROVED** [review]

**Pull request merged**